crates/nu-parser/src/parser.rs

```diff
@@ -71,7 +71,12 @@ pub fn is_math_expression_like(
         return false;
     }
 
-    if bytes == b"true" || bytes == b"false" || bytes == b"null" || bytes == b"not" {
+    if bytes == b"true"
+        || bytes == b"false"
+        || bytes == b"null"
+        || bytes == b"not"
+        || bytes == b"if"
```

Comment:
Go to the implementation detail...treating if as a mathematical expression feels counterintuitive.  Not really sure should we need to rename the function?

Reply:
Yeah, the function and shape should probably get a different name when we're ready to take this in.